nilearn/decoding/tests/test_decoder.py

```diff
@@ -87,18 +87,30 @@ def test_check_inputs_length():
                                         screening_percentile=100.).fit, X_, y)
 
 
-def test_check_estimator():
-    iris = load_iris()
-    X, y = iris.data, iris.target
-    y = 2 * (y > 0) - 1
-    X_, mask = to_niimgs(X, (2, 2, 2))
+def test_BaseDecoder_check_estimator():
+    # Check if the estimator is one of the supported estimators, and if not,
+    # if it is a string, and if not, then raise the error
 
-    for estimator in ['log_l1', 'log_l2', 'ridgo']:
-        assert_raises(ValueError, BaseDecoder(estimator=estimator).fit, X_, y)
+    supported_estimators = ['svc', 'svc_l2', 'svc_l1', 'logistic',
+        'logistic_l1', 'logistic_l2', 'ridge', 'ridge_classifier',
+        'ridge_regressor', 'svr']
+    
+    wrong_estimators = ['ridgo', 'svb']
+
+    for estimator in supported_estimators:
+        BaseDecoder(estimator=estimator)._check_estimator()
```

Comment:
Nothing is being checked here.
I believe this will generate an unknown estimator warning even with supported estimators.